go.mod

```diff
@@ -4,9 +4,9 @@ go 1.16
 
 require (
 	github.com/filecoin-project/go-data-transfer v1.10.1
-	github.com/filecoin-project/go-indexer-core v0.1.5
+	github.com/filecoin-project/go-indexer-core v0.2.0
 	github.com/filecoin-project/go-legs v0.0.0-20211002011713-f1f52b88c467
-	github.com/filecoin-project/storetheindex v0.0.0-20211001170917-11797057b28d
+	github.com/filecoin-project/storetheindex v0.0.0-20211008013136-08d087cd7027
```

Comment:
ideally we don't need to depend on the 'storetheindex' as well as 'go-indexer-core' we should understand where the dep came from.

Reply:
The dependency on go-indexer-core is for data type definitions, such as `indexer.Value` and `indexer.Metadata`.  The dependency on storetheindex is due to interface and API definitions.

It should be possible to only depend on storetheindex by removing the use `indexer.Value` (which should only be used between storetheindex and go-indexer-core), and by moving the definition of `indexer.Metadata` into storetheindex.  The second part doable since go-indexer-core only cares about the `indexer.Value` and does not need to see the metadata structure.

Reply:
@willscott This is address by https://github.com/filecoin-project/go-indexer-core/pull/30 and  https://github.com/filecoin-project/storetheindex/pull/93, and will be addressed by a subsequent PR in this repo.